.github/workflows/build-and-test.yml

```
@@ -1,24 +1,30 @@
-name: build_test
+name: build-and-test
+
 on:
-  workflow_dispatch:
-  pull_request:
-    types: [opened, synchronize]
+  workflow_call:
+    inputs:
+      build-all-platforms:
+        description: 'Build wheels for all platforms (Linux ARM64, Windows, macOS)'
+        required: false
+        type: boolean
+        default: true
+
```

I'll be changing the default permission to read-only, so we need to add write capability, otherwise cache push won't work
```suggestion
permissions:
  contents: read
  actions: write
```